table/sst_file_dumper.cc

```diff
@@ -125,17 +125,31 @@ Status SstFileDumper::GetTableReader(const std::string& file_path) {
                               nullptr);
       file_.reset(new RandomAccessFileReader(std::move(file), file_path));
     }
-    options_.comparator = &internal_comparator_;
+
     // For old sst format, ReadTableProperties might fail but file can be read
     if (ReadTableProperties(magic_number, file_.get(), file_size,
                             (magic_number == kBlockBasedTableMagicNumber)
                                 ? &prefetch_buffer
                                 : nullptr)
             .ok()) {
       s = SetTableOptionsByMagicNumber(magic_number);
+      if (s.ok()) {
+        if (table_properties_ != nullptr &&
```

Comment:
According to c++20 (https://en.cppreference.com/w/cpp/memory/unique_ptr), we should avoid doing `!=` for unique_ptr. Although RocksDB is still C++11, and we are in the middle of moving to C++17, we would like to make sure we are not using features unavailable in more recent standards.
IN this case, we can use `if (table_properties_`

Reply:
addressed this.